### DISCUSSION TITLE
What're the 3D models' requirements for AR ?

### DISCUSSION THREAD

**Discussion by vaggarath** [google/model-viewer#3407] [Q&A] [answered]
I prefer to ask my noob question here for it isn't an issue from model viewer but more on my end with the model i'm using.

The problem i'm facing is that the models we use are either unable to appear at all on AR mode or they appear hugely and it's a true pain the neck to manipulate them. While, with the same JS code (i made a class which generate the modelviewer itself along with its options), i have absolutely no problem using some low poly models in AR, like the astronaut by exemple.

So i'm wondering what's wrong with our models. Does the weight plays a role in this ? (between 50 & 150 Mo each - yeah they ain't featherleight^^). 

Is it the number of triangles/faces ? Ours are a cute 2Millions for a light one (yeaaaaah....^^)

Does the original position of the camera in the GLTF model has anything to do with it ? (usually models swallows the camera when imported in blender)

Could the scale play a part ? When i load a model on blender the model seems huge compared to models like the astronaut or the ones from khronos.

Doest the original position of the model itself as a role to play ? (i'm more than a noob about 3D modeling but i saw that when i open our models on blender they're, on the "plane", half below half upper as when i opened the astronaut model you're displaying it's only on the upper surface of what i stupidly call a plane.

Again, thanks for any advices / explanation anyone could give me as i'm such a low noob on this! I'm sorry if i'm not always clear or have bad english, but if further details are needed i'll be more than happy to oblige.
![ex_frelon](https://user-images.githubusercontent.com/44065516/166206826-881b8b1a-34aa-4114-a760-6649e994184d.png)

I attached a screenshot that show a model scale and also the camera position when i just import it

**Comment by elalish** [comment, marked as answer]
Potentially each of the things you mention can be an issue and are worth looking into, but for the problem you describe I believe the primary culprit is scale. glTF specifies that all models are in meters, which is what allows us to put them at true scale in AR. Likely the fix is as simple as adding a top-level transform to put them at their desired size in meters (probably not actual ant-sized, since that would also be hard to see in AR, but rather your desired scale presentation size).

**Reply by elalish** [reply]
You can also do this much quicker by setting the [`scale`](https://modelviewer.dev/docs/#entrydocs-scenegraph-attributes-scale) attribute in `<model-viewer>`.

**Comment by vaggarath** [comment]
thanks ! I'm ashamed, i specifically looked for a scale option and never saw it at any point >< (even worse is that it's with the orientations exemples which i used for keyboard controls -_-' )

But your explanation says it all. That's why in AR the hornet or even the tarentula were each the size of several cars and working only in wide spaces like outdoors. 

For the scale it seems to actually do the tricks except for one model but again it's from our end so i'll ask the colleague to check on it.

Thanks again for taking time to explain !
